de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetBetaWarning.java

```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.uiengine.ui.editor.snippets.general;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.QualifiedName;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.ISelectionChangedListener;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+
+import de.dlr.sc.virsat.external.lib.commons.cli.Activator;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.AUiSectionSnippet;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet;
+
+/**
+ *
+ * A snippet that warns if the current element is a BETA element or contains one
+ *
+ */
+public class UiSnippetBetaWarning extends AUiSectionSnippet implements IUiSnippet {
+
+	private static final String SECTION_HEADING = "Beta Warning!";
+	private static final int UI_LAYOUT_NR_COLUMNS = 1;
+	private static final String DEMO_ACTIVE = "demo";
+
+	private final boolean isDemoActive;
+
+	/**
+	 * Constructor for the beta warning snippet
+	 */
+	public UiSnippetBetaWarning() {
+		super();
+		isDemoActive = Activator.getCommandLineManager().isCommandLineOptionSet(DEMO_ACTIVE);
+	}
+	
+	@Override
+	public void createSwt(FormToolkit toolkit, EditingDomain editingDomain, Composite composite, EObject initModel) {
+		super.createSwt(toolkit, editingDomain, composite, initModel);
+
+		Composite sectionBody = createSectionBody(toolkit, SECTION_HEADING, null, UI_LAYOUT_NR_COLUMNS);
+
+		Label label = toolkit.createLabel(sectionBody, createDescriptionLabel());
+
+		Display display = Display.getCurrent();
+		Color red = display.getSystemColor(SWT.COLOR_DARK_RED);
+		Color gray = display.getSystemColor(SWT.COLOR_GRAY);
+		sectionBody.setBackground(red);
+		label.setBackground(red);
+		label.setForeground(gray);
+		
+		setUpLabel(label);
+

```

Comment:
Maye remove unwated empty lines. Looks a bit cleaner i would say.

Reply:
It's not unwanted... It separates the color operations from the generic setup method